lib/synthesize-stream.ts

```diff
@@ -126,17 +126,42 @@ class SynthesizeStream extends Readable {
 
     socket.onmessage = message => {
       const chunk = message.data;
-      // some messages are strings - emit those unencoded, but push them to
-      // the stream as binary
-      const data = typeof chunk === 'string' ? chunk : Buffer.from(chunk);
+      // some info messages are sent as strings, telling the content_type and
+      // timings. Emit them as separate events, but do not send them along the
+      // pipe.
+      if (typeof chunk === 'string') {
+        try {
+          const json = JSON.parse(chunk);
+          if (json['binary_streams']) {
+            self.emit('binary_streams', message, chunk);
+          }
+          else if (json['marks']) {
+            self.emit('marks', message, chunk);
+          }
+          else if (json['words']) {
+            self.emit('words', message, chunk);
+          }
+          else if (json['error']) {
+            self.emit('error', message, chunk);
+          }
+          else if (json['warnings']) {
+            self.emit('warnings', message, chunk);
+          }
+        }
+        finally {
+          self.emit('message', message, chunk);
```

Comment:
I wanted to keep things consistent with the prior method of how things were outputted, but for the individual channels, may make more sense to emit like `self.emit('<channel>', json, message);`

Reply:
I hear what you're saying but I think we should keep this consistent for now. Returning the whole response in the first arg is flexible since we will probably always want to return that. I don't think it would be a bad idea to pass `json` instead of `chunk` here, though, to give the user an already-parsed object.

Reply:
Agreed, I've updated the messages that could emit `json` to send that instead, and that all will output a string on `message` to retain BC.